app/models/concerns/user_encrypted_email_overrides.rb

```diff
@@ -0,0 +1,59 @@
+module UserEncryptedEmailOverrides
+  extend ActiveSupport::Concern
+
+  # override some Devise methods to support our use of encrypted_email
+
+  class_methods do
+    def find_first_by_auth_conditions(tainted_conditions, opts = {})
+      if tainted_conditions[:email].present?
+        opts[:email_fingerprint] = create_fingerprint(tainted_conditions.delete(:email))
+      end
+      to_adapter.find_first(devise_parameter_filter.filter(tainted_conditions).merge(opts))
+    end
+
+    def find_with_email(email)
+      email_fingerprint = create_fingerprint(email)
+      find_by(email_fingerprint: email_fingerprint)
+    end
+
+    def create_fingerprint(email)
+      Pii::Fingerprinter.fingerprint(email.downcase)
+    end
+  end
+
+  def postpone_email_change?
```

Comment:
Is it worth commenting what this is doing? I'm not intimately familiar with Devise's internal workings, and we seem to be overriding the entire method to add one or two conditions to the if statement?

Also in case devise ever changes and we need to re-do this patch?